metricbeat/module/kubernetes/util/metrics_storage.go

```diff
@@ -0,0 +1,200 @@
+package util
+
+import (
+	"strings"
+	"sync"
+)
+
+type Metrics struct {
+	entries map[Metric]float64
+}
+
+func NewMetrics() *Metrics {
```

Comment:
Since this "Storage" functionality is new it would worth documenting it within the same file. How it works, some examples and how the structs/objects are linked to each other. This would make this mechanism readable and even re-usable in the future.

Reply:
Do you mean adding some comments to explain all of this in the same file?

Reply:
Yes, whatever that would make the usage of this new mechanism clear to follow/understand.